### PR TITLE
Parsing /status comand

### LINKED ISSUE #34: [Dead By Design] status command for single scenario is not working
Current implementation splitting bot command by spaces can't parse command like `/status awx:Prepare test host`

### PR DESCRIPTION
**Pull request by Polina-Gubina**
Command is separated only spaces now (e.g. `/status awx "Scenario 1.5" 2`)

Spaces are not taken in account if used in quotes (`"..."` or `'...'`)

Close #34

### CODE REVIEW

File: delatore/bot/parsing.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if not arguments:
+        return
+    arguments[0] = arguments[0].replace(' ', '')
+    target: str = arguments[0].split(":")
+    name_template = None
+    count = 1
+    if target[0] == 'awx':
+        if len(arguments) > 1:
+            name_template = arguments[1].strip()
+            if (len(arguments) > 2) & (arguments[2] != ''):
```
Comment (outcatcher) [inline delatore/bot/parsing.py]: Use `and` for logical "and"

File: delatore/bot/parsing.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                arguments[2] = arguments[2].replace(" ","")
```
Comment (outcatcher) [inline delatore/bot/parsing.py]: Better not to update arguments list. `str.strip` seems to be more appropriate method in this place

File: tests/test_parsing.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+def test_parse_quotted_message(quo):
+    expected = ('awx', 'Scenario 1.5', 5)
+    message = f'{expected[0]}:\'{expected[1]}\'{expected[2]}'
+    source, detailed, count = parse_command(message)
+    assert (source, detailed, count) == expected
+
+
+@pytest.mark.parametrize(('message', 'expected'),
+                         [
+                             ('/status awx:\'Destroy\' 1', ('awx', 'Destroy', 1)),
```
Comment (outcatcher) [inline tests/test_parsing.py]: Add test with non-quoted target

File: delatore/bot/parsing.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+__all__ = ["parse_command"]
+
+
+def parse_command(message):
+    if message.startswith('/status'):
+        message = message [len('/status'):]
+    return _parse_status(message)
+
+
+def _parse_status(message):
```
Comment (outcatcher) [inline delatore/bot/parsing.py]: It seems to be better to use `shlex` for correct quoted string handling

File: tests/test_parsing.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import pytest
+
+from delatore.bot.parsing import parse_command
```
Comment (anton-sidelnikov) [inline tests/test_parsing.py]: make import relative

**Comment by outcatcher** [comment]
Please also update readme, removing `:` usage in commands

File: README.md [hunk 6]
```diff
@@ -31,14 +31,15 @@ Bot reply to the message with last status(-es) retrieved from given source
 
 Status has following syntax:
 
-`/status <source>[:detailed_source] [history_depth]`
+`/status <source> '[detailed_source]' [history_depth]`
```
Comment (outcatcher) [inline README.md]: Quotes are not required and not a part of the command

File: README.md [hunk 7]
```diff
@@ -31,14 +31,15 @@ Bot reply to the message with last status(-es) retrieved from given source
...
 
 #### AWX Source
 
 Status command for AWX source has following syntax:
 
-`/status awx[:template_name] [history_depth]`
+`/status awx '[template_name]' [history_depth]`
```
Comment (outcatcher) [inline README.md]: See previous comment

File: delatore/bot/main.py [hunk 8]
```diff
@@ -85,23 +87,18 @@ def start(self):
...
-        response = AWXApiClient().create_status_message(template)
+    try:
+        source, template_name, count = parse_command(message.text)
+    except CommandParsingError:
+        response = bot.reply_to(message, "Invalid command. Please check command syntax")
+        return bot.alert(message.chat.id, response)
+    try:
+        response = AWXApiClient().create_status_message(template_name)
         bot.alert(message.chat.id, response)
+    except:
```
Comment (outcatcher) [inline delatore/bot/main.py]: Exception should be raised inside of `create_status_message` and be more precise. `except` is catching all exceptions

File: delatore/bot/parsing.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+class CommandParsingError(Exception):
+    """Error during command parsing"""
+
+
+def parse_command(message):
+    """Parsing command arguments to arguments list"""
+    LOGGER.debug('Got message: %s', message)
+    cmd, target, *args = shlex.split(message)
+    depth = None
+    if  ("\'" not in message) and ("\"" not in message) and (len(args) > 1):
```
Comment (outcatcher) [inline delatore/bot/parsing.py]: single quotes can be used

File: delatore/sources/awx_api.py [hunk 10]
```diff
@@ -44,7 +48,10 @@ def create_status_message(self, template: str = None):
         :param template: name of template, if empty — all scenarios
         """
         _filter = {'name__iexact': template}
-        json_data = self.get_templates(_filter)
+        try:
+         json_data = self.get_templates(_filter)
```
Comment (outcatcher) [inline delatore/sources/awx_api.py]: should be 4 spaces, not 1

**Pull request merged**